CMakeLists.txt

```diff
@@ -162,9 +163,11 @@ if(OpenMP_FOUND)
     target_link_libraries(gtda_simplex_tree PRIVATE OpenMP::OpenMP_CXX)
 endif()
 
+target_link_libraries(gtda_simplex_tree LINK_PUBLIC Eigen3::Eigen)
```

Comment:
Most of my interventions on `CMakeLists.txt` are guesswork. @MonkeyBreaker please check carefully :)